private/bufpkg/bufmodule/bufmodule.go

```diff
@@ -472,7 +473,37 @@ func ModuleToBucket(
 			return err
 		}
 	}
-	return bufmoduleref.PutDependencyModulePinsToBucket(ctx, writeBucket, module.DependencyModulePins())
+	if err := bufmoduleref.PutDependencyModulePinsToBucket(ctx, writeBucket, module.DependencyModulePins()); err != nil {
+		return err
+	}
+	// This is the default version created by bufconfig getters. The versions should be the
+	// same across lint and breaking configs.
+	version := bufconfig.V1Beta1Version
+	var breakingConfigVersion string
+	if module.BreakingConfig() != nil {
+		breakingConfigVersion = module.BreakingConfig().Version
+	}
+	var lintConfigVersion string
+	if module.LintConfig() != nil {
+		lintConfigVersion = module.LintConfig().Version
+	}
+	if breakingConfigVersion != "" && lintConfigVersion != "" {
+		if breakingConfigVersion != lintConfigVersion {
+			return fmt.Errorf("breaking config version %q does not match lint config version %q", breakingConfigVersion, lintConfigVersion)
+		}
+		version = breakingConfigVersion
+	} else if breakingConfigVersion != "" {
+		version = breakingConfigVersion
+	} else if lintConfigVersion != "" {
+		version = lintConfigVersion
+	}
```

Comment:
We should return an error in these cases. We should force the caller of this function to provide consistent versions for the module's `LintConfig` and `BreakingConfig`, otherwise it's an internal error on our end since that should always be in our control.

We should also be able to guarantee that the `{Breaking,Lint}Config` version is _always_ non-empty if the `{Breaking,Lint}Config` is non-`nil`.

So in short:
* A version _must_ be set on both of the `LintConfig` and `BreakingConfig` types if they are non-`nil`.
* The version _must_ be equivalent between them if they are both non-`nil`.
* If one of the `{Breaking,Lint}Config` types is `nil`, the versions don't necessarily need to match, but this isn't actually possible in practice because there's always a default value for each one. If one of the configs is provided, both will be non-`nil`.